src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -574,6 +574,34 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
     val runsRightAfter = None
   } with GenBCode
 
+  // phaseName = "analyzeDeps"
+  object analyzeDeps extends {
+    val global: Global.this.type = Global.this
+  } with SubComponent {
+    val phaseName = "analyzeDeps"
+    val runsAfter = List("jvm")
+    val runsRightAfter = None
+
+    def newPhase(prev: Phase): GlobalPhase = {
+      new AnalyzeDepsPhase(prev)
+    }
+    private class AnalyzeDepsPhase(prev: Phase) extends GlobalPhase(prev) {
```

Comment:
I suspect that this phase could be implemented without needng to modify the compiler to keep track of `completedClassfiles`.  We seem to have all the information we need with a walk of the symbol table.

Here's a prototype: https://github.com/retronym/scalac-survival-guide/blob/master/src/main/scala/guide/_23_CompletedSymbols.scala

That outputs:

https://gist.github.com/retronym/0abbd4ab969f2743ddf2ef30054bf0d4

This would mean you could distribute it as a compiler plugin, which would help it evolve into a useful/mature tool before adding it to the compiler itself, and would help people on older compiler versions to use it.